concordium-consensus/src/Concordium/Queries.hs

```diff
@@ -59,55 +60,73 @@ import qualified Concordium.GlobalState.TreeState as TS
 import Concordium.GlobalState.Types
 import qualified Concordium.GlobalState.Wasm as GSWasm
 import Concordium.ID.Types
+import qualified Concordium.KonsensusV1.Consensus as ConsensusV1
+import qualified Concordium.KonsensusV1.SkovMonad as SkovV1
+import qualified Concordium.KonsensusV1.TreeState.Implementation as SkovV1
+import qualified Concordium.KonsensusV1.TreeState.Types as SkovV1
+import qualified Concordium.KonsensusV1.Types as SkovV1
 import Concordium.Kontrol
 import Concordium.Kontrol.BestBlock
 import Concordium.MultiVersion
 import Concordium.Skov as Skov (
     SkovQueryMonad (getBlocksAtHeight),
     evalSkovT,
  )
+import Control.Monad.State.Class
+import Data.Time
+
+-- |Type of a query that can be run against consensus version 0.
```

Comment:
I have created a task for going over the FIXME's for a subsequent PR. https://github.com/Concordium/concordium-node/issues/857

Reply:
Added references.